tests/test_s3_utils.py

```diff
@@ -46,7 +59,77 @@ def should_successfully_get_object_from_staging_bucket() -> None:
         key=f"{key_prefix}/{collection_metadata_filename}",
     ):
 
-        s3_url_reader = get_s3_url_reader(get_s3_role_arn())
-        json_object = load(s3_url_reader(collection_metadata_url))
+        s3_url_reader = get_s3_url_reader(get_s3_role_arn(), any_dataset_prefix(), get_log())
+        json_object = load(s3_url_reader(collection_metadata_url).response["Body"])
+
+        assert json_object == collection_dict
+
+
+@mark.infrastructure
+def should_get_object_from_storage_bucket_when_staging_bucket_fails() -> None:
```

Comment:
"when_staging_bucket_fails" could maybe be clarified as "when_not_in_staging_bucket"?